### PR TITLE
Clean up database dumping process

### PR DESCRIPTION
**Pull request by vlaaad**
Previous state of database dumping was confusing to me. I worry there might be
a good motivation to keep old dumps around, although it seems that a lot of
old dumps around are unused. I removed all the unused dumps, keeping only
bi-directional json mapping from and to new ids.

The old way of db dumping used taxonomy_to_concept_v67.json file as a source of
truth for some ids that were dumped from datomic. I removed it, since data in
the newest database to get dumps from (jobtech-taxonomy-development-2020-03-18-2)
is matching the data in v67.

Previous dumping process also renamed all types to kebab-case, which is
effectively noop on current types, so I removed that as well: it seems to be a
relic of some old schema.

### CODE REVIEW

File: src/jobtech_taxonomy_database/converters/nano_id_assigner.clj [hunk 1]
```diff
@@ -1,56 +1,11 @@
...
-  "Compare taxonomy-ID from sql and taxonomy-ID from json"
-  [category-67 id-67]
-  (let [id (find-id category-67 id-67)]
-    (if id id "no nano-id")))
-
-(defn compare-descriptions
-  "Compare concept description from sql and concept description from json"
-  [category-67 id-67 description-67]
-  (let [description (find-description category-67 id-67)]
-    (if (= description-67 description) "same description" "different description")))
```
Comment (vlaaad) [inline src/jobtech_taxonomy_database/converters/nano_id_assigner.clj]: Removed a lot of unused code here

File: src/jobtech_taxonomy_database/utils/merge_new_nano_ids.clj [hunk 2]
```diff
@@ -1,54 +0,0 @@
...
-  "Open json, return map with json keyword formatted as clojure keywords."
-  [fname-in]
-  (parse-string (slurp fname-in) true))
-
-(defn -main
-  []
-  (let* [org-struct  (open-json "resources/taxonomy_to_concept_v67.json")
-         result (merge-new-ids org-struct "resources/ids-for-skill-headlines-and-mainheadlines_v67.json")]
-        (save (generate-string result {:pretty true}) "resources/taxonomy_to_concept_with_headlines_and_mainheadlines_v67.json")
-        result))
```
Comment (vlaaad) [inline src/jobtech_taxonomy_database/utils/merge_new_nano_ids.clj]: This file was using non-existing resources `resources/ids-for-skill-headlines-and-mainheadlines_v67.json` and didn't work, so I removed it.

File: src/jobtech_taxonomy_database/converters/nano_id_assigner.clj (hunk 1 above)
Comment (weipe) [reply src/jobtech_taxonomy_database/converters/nano_id_assigner.clj]: Good!

File: src/jobtech_taxonomy_database/dump.clj [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+(defn dump-taxonomy-to-json []
+  (let [[concept->taxonomy taxonomy->concept]
+        (->> (d/q find-concept-by-query (get-db))
+             (map first)
+             (map rename-concept-keys-for-json)
+             (reduce (fn [[acc-concept->taxonomy acc-taxonomy->concept] concept]
+                       [(assoc acc-concept->taxonomy (:conceptId concept) (dissoc concept :conceptId))
+                        (assoc-in acc-taxonomy->concept [(:type concept) (:legacyAmsTaxonomyId concept)] concept)])
+                     [{} {}])
+             (walk/postwalk #(cond->> % (map? %) (into (sorted-map)))))]
```
Comment (weipe) [inline src/jobtech_taxonomy_database/dump.clj]: walk/postwalk, elegant!

File: src/jobtech_taxonomy_database/utils/merge_new_nano_ids.clj (hunk 2 above)
Comment (weipe) [reply src/jobtech_taxonomy_database/utils/merge_new_nano_ids.clj]: Very good!

**Comment by weipe** [comment]
Good work! Yes, they types were once snake-cased, and renamed to kebab case.

**Pull request merged**